V5 x

mohammadsalem opened the pull request.

mohammadsalem committed:
Index the last 12 months vies and downloads for each item and expose them to the API with an option to get statistics for a selected period (1 month - 12 months)

mohammadsalem committed:
v5.2.0
- Query solr directly without indexing into PostgreSQL database.
- Possibility to aggregate by country/city and period (monthly).

mohammadsalem committed:
Downloads query fix.

mohammadsalem committed:
Cleanup

mohammadsalem committed:
Single item downloads query fix.

alanorth commented:
Exciting! My first feedback is that the `limit` parameter is broken: with the default limit of 100 I see 191 items on the first page. If I set the limit to 1 I get 2. Something is up there.

While you're thinking about that I want to try to benchmark the Solr performance if we query all items for multiple years of data on the fly. We should probably use a transparent requests cache in my opinion. I've done that in a [few scripts that hit remote APIs](https://github.com/ilri/DSpace/blob/5_x-prod/agrovoc-lookup.py#L80) and it's super easy and performant.

mohammadsalem committed:
Get downloads for the items returned from the views query.
Return total_views_by_month and total_downloads_by_month for the whole date period even if no statistics exists.

mohammadsalem committed:
For single item, get downloads independently from the views.

mohammadsalem commented:
@alanorth True! it was getting 100 views and 100 downloads then combine them, as they are not stored in the same order it resulted more than 100 items, also this was resulting some items to be returned without views or downloads as they appear in one query but not the other.

Now it will get the downloads only for items returned from the views query. This will fix it but will make another issue! If an item has only downloads it will never appear.
While for single item everything should be fine.

alanorth commented:
No activity on v5 branch now. Only working with v6 and above.
